### PR TITLE
♻️ 🧪 Introduces an e2e test methodology

### PR DESCRIPTION
**Pull request by ccamel**
## Purpose

This PR introduces an e2e test methodology based on [Gherkin](https://cucumber.io/docs/gherkin/), a textual, behavior-oriented approach that is human-readable, simple to implement, and highly extensible. This is the same approach adopted for the [axone-protocol/axoned](https://axone-protocol/axoned) repository as seen in [this PR](https://github.com/axone-protocol/axoned/pull/579).

## Objective

This approach has several aims. First and foremost, while our test coverage is currently excellent, it suffers from a number of weaknesses:
- The Tester Experience (TX) is suboptimal because writing tests in Rust is difficult to implement, develop, and maintain, especially regarding the complexity of some data types (like the ones used for querying).
- The secondary objective of serving as a functional repository and documentation is not fully realized due to the complexity of Rust-based tests.

Gherkin's approach addresses these issues by focusing on the feature itself and introducing a DSL that simplifies the process of writing and understanding tests. This DSL allows for more readable and maintainable tests, improving both the implementation and documentation aspects of our project.

## Changes

To pave the way, this PR introduces a very basic [Gherkin](https://cucumber.io/docs/gherkin/) test suite that allows for the execution of `InstantiateMsg` and `ExecuteMsg` for the `InsertData` operation onto the `Cognitarium` Smart Contract. This lays the groundwork for a more user-friendly and maintainable testing framework I believe.

## Summary by CodeRabbit

- **New Features**
  - Introduced a feature for inserting RDF data into the Cognitarium smart contract.
  - Added behavior-driven development testing using Cucumber for smart contracts.

- **Tests**
  - Enhanced testing framework with new dependencies and configurations for Cucumber.
  - Implemented comprehensive tests for smart contracts, including scenario parsing and step definitions.

- **Refactor**
  - Restructured module declarations for better organization.
  - Removed certain code attributes to streamline codebase.

### CODE REVIEW

**Review by bdeneux — APPROVED** [review]
Wonderful 🤩

File: contracts/axone-cognitarium/Cargo.toml [hunk 1]
```diff
@@ -35,10 +35,26 @@ thiserror.workspace = true
...
+harness = false
+name = "cucumber"
+path = "tests/e2e/main.rs"
+
+[package.metadata.scripts]
+optimize = """docker run --rm -v "$(pwd)":/code \
+  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+  cosmwasm/rust-optimizer:0.12.10
+"""
```
Comment (amimart) [inline contracts/axone-cognitarium/Cargo.toml:60]:
```suggestion
```

This shouldn't be needed I think :)

**Review by amimart — CHANGES_REQUESTED** [review]
Looks pretty good to me, thx 🙏 
Can't wait to feed the documentation with that 😉

File: contracts/axone-cognitarium/Cargo.toml (hunk 1 above)
Comment (ccamel) [reply contracts/axone-cognitarium/Cargo.toml]: Oh yes, absolutely! 😅 Thanks for spotting it.
Fixed with 1771bb5.

**Review by amimart — APPROVED** [review]

**Pull request merged**